crates/bevy_ecs/src/system/system.rs

```diff
@@ -24,10 +17,11 @@ pub trait System: Send + Sync + 'static {
     type Out;
     fn name(&self) -> Cow<'static, str>;
     fn id(&self) -> SystemId;
-    fn update(&mut self, world: &World);
+    fn update_access(&mut self, world: &World);
     fn archetype_component_access(&self) -> &TypeAccess<ArchetypeComponent>;
+    fn component_access(&self) -> &TypeAccess<TypeId>;
```

Comment:
`archetype_component_access` is filled from archetypes, I can't extract just the types without impacting its performance or function; `component_access` is used just once (per inserted system), while `archetype_component_access` is invoked every time archetypes change. This will likely be revised later, when storage changes land, so it's good enough for now.

Reply:
Oh, it occurred to me that the question might be "why this exists at all": because we want to know about ambiguities statically. We only check for them once, potentially when no archetypes have been created at all, let alone all of those that might ever exist in the app, so we have to assume that any and all combinations will exist - checking just component types as if archetypes didn't exist achieves that.